packages/salesforcedx-vscode-apex/src/apexLanguageClient.ts

```diff
@@ -21,4 +31,62 @@ export class ApexLanguageClient extends LanguageClient {
   public get errorHandler(): ApexErrorHandler | undefined {
     return this._errorHandler;
   }
+
+  public async stop(): Promise<void> {
+    await super.stop();
+  }
+
+  public async showOrphanedProcessesDialog(orphanedProcesses: ProcessDetail[]) {
+    const orphanedCount = orphanedProcesses.length;
+
+    if (orphanedCount === 0) {
+      return;
+    }
+
+    let choice: string | undefined = nls.localize('terminate_show_processes');
+    do {
+      choice = await vscode.window.showWarningMessage(
+        nls.localize(
+          'terminate_orphaned_language_server_instances',
+          orphanedCount
+        ),
+        nls.localize('terminate_processes'),
+        nls.localize('terminate_show_processes')
+      ) ?? 'dismissed';
+
+      if (choice === nls.localize('terminate_processes') && await terminationConfirmation(orphanedCount)) {
+        for (const processInfo of orphanedProcesses) {
+          try {
+            await terminateProcess(processInfo.pid);
+            channelService.appendLine(nls.localize('terminated_orphaned_process', processInfo.pid));
+          } catch (err) {
+            channelService.appendLine(nls.localize('terminate_failed', processInfo.pid, err.message));
+          }
```

Comment:
A telemetry addition here would be _awesome_.  Both for the case where we kill all the processes successfully and on for reporting the error if we fail.